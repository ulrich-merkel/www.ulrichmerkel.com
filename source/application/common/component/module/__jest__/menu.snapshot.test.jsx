/* global jest, describe, expect */
/* eslint-disable react/prefer-es6-class, react/prefer-stateless-function, func-names, react/prop-types, import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
import ComponentToBeTested from './../menu';

// @see https://github.com/airbnb/enzyme/issues/426
// @see https://github.com/facebook/jest/issues/1353
describe('component/module/menu', function () {
    describe('Snapshot', function () {
        it('should render correctly', function () {
            var tree = renderer.create(
                <ComponentToBeTested
                    componentType='span'
                    className='foo'
                    content={{
                        name: 'bar',
                        list: [
                            {
                                url: '/foo',
                                headline: 'headline',
                                lead: 'lead',
                                img: {}
                            },
                            {
                                url: '/foo',
                                headline: 'headline',
                                lead: 'lead',
                                img: {}
                            },
                            {
                                url: '/foo',
                                headline: 'headline',
                                lead: 'lead',
                                img: {}
                            }
                        ]
                    }}
                >
                    Hello
                </ComponentToBeTested>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
