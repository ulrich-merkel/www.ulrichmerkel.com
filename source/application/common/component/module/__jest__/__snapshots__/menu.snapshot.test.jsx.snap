exports[`component/module/menu Snapshot should render correctly 1`] = `
<span
  className="m-menu m-menu--bar foo"
  itemScope={true}
  itemType="https://schema.org/ItemList"
  role="menu" />
`;
